packages/react-storage/src/components/StorageBrowser/actions/configs/types.ts

```diff
@@ -80,63 +84,93 @@ export interface TaskActionConfig<T extends TaskHandler> {
+> extends TaskActionConfig<T, ComponentName> {
   /**
    * sets action view subcomponent type and action processing behavior
    */
-  type: 'BATCH_ACTION' | 'SINGLE_ACTION';
+  type: 'BATCH_ACTION';
 }
 
-export interface ListActionConfig<T> extends ActionConfigTemplate<T> {}
+export interface SingleTaskActionConfig<
+  T extends TaskHandler,
+  ComponentName extends Capitalize<string> = Capitalize<string>,
+> extends TaskActionConfig<T, ComponentName> {
+  /**
+   * sets action view subcomponent type and action processing behavior
+   */
+  type: 'SINGLE_ACTION';
+}
 
-export interface UploadActionConfig extends TaskActionConfig<UploadHandler> {}
+export interface ListActionConfig<T, ComponentName>
+  extends ActionConfigTemplate<T> {
+  componentName: ComponentName;
+}
+
+export interface UploadActionConfig
+  extends TaskActionConfig<UploadHandler, 'UploadView'> {
+  type: 'BATCH_ACTION';
+  isCancelable: IsCancelableTaskHandler<UploadHandler>;
+}
 
 export interface CreateFolderActionConfig
-  extends TaskActionConfig<CreateFolderHandler> {}
+  extends TaskActionConfig<CreateFolderHandler, 'CreateFolderView'> {
+  type: 'SINGLE_ACTION';
+  isCancelable: IsCancelableTaskHandler<CreateFolderHandler>;
+}
 
 export interface ListLocationsActionConfig
-  extends ListActionConfig<ListLocationsHandler> {
-  title: string;
+  extends ListActionConfig<ListLocationsHandler, 'ListLocationsView'> {
+  displayName: string;
   type: 'LIST_LOCATIONS';
 }
 
 export interface ListLocationItemsActionConfig
-  extends ListActionConfig<ListLocationItemsHandler> {
-  title: (bucket: string | undefined, prefix: string | undefined) => string;
+  extends ListActionConfig<ListLocationItemsHandler, 'ListLocationItemsView'> {
+  displayName: (
+    bucket: string | undefined,
+    prefix: string | undefined
+  ) => string;
   type: 'LIST_LOCATION_ITEMS';
 }
 
 export interface DefaultActionConfigs {
-  listLocationItems: ListLocationItemsActionConfig;
-  listLocations: ListLocationsActionConfig;
-  createFolder: CreateFolderActionConfig;
-  upload: UploadActionConfig;
+  ListLocationItems: ListLocationItemsActionConfig;
+  ListLocations: ListLocationsActionConfig;
+  CreateFolder: CreateFolderActionConfig;
+  Upload: UploadActionConfig;
 }
 
-export type DefaultActionKey = keyof DefaultActionConfigs;
-export type CustomActionKey<T> = keyof Omit<T, DefaultActionKey>;
+export type DefaultActionKeys = keyof DefaultActionConfigs;
 
-export type ActionConfigs = Record<
-  Capitalize<string>,
-  | TaskActionConfig<TaskHandler>
+export type ActionConfigs<
+  ActionsKeys extends Capitalize<string> = Capitalize<string>,
+> = Record<
+  ActionsKeys,
+  | BatchTaskActionConfig<TaskHandler>
+  | SingleTaskActionConfig<TaskHandler>
   | ListLocationItemsActionConfig
   | ListLocationsActionConfig
 >;
 
 export type ResolveActionHandler<T> = T extends
   | TaskActionConfig<infer K>
-  | ListActionConfig<infer K>
+  | ListActionConfig<infer K, string>
```

Comment:
Can we default the second positional generic of `ListActionConfig` to `Capitalize<string>` to keep alignment with `TaskActionConfig`?